File: commonfamily/src/main/java/com/koenig/communication/messages/family/CreateUserMessage.java
package com.koenig.communication.messages.family;

import com.koenig.commonModel.Byteable;
import com.koenig.commonModel.Component;
import com.koenig.communication.messages.FamilyMessage;

import org.joda.time.DateTime;

import java.nio.ByteBuffer;

/**
 * Created by Thomas on 11.10.2017.
 */

public class CreateUserMessage extends FamilyMessage {

    public static final String NAME = "CreateUserMessage";

    private String userName;
    private DateTime birthday;

    public CreateUserMessage(String userName, DateTime birthday) {
        super(Component.FAMILY);
        this.birthday = birthday;
        this.userName = userName;
    }

    public CreateUserMessage(int version, Component component, String fromId, String toId, ByteBuffer buffer) {
        super(version, component, fromId, toId);
        userName = Byteable.Companion.byteToString(buffer);
        birthday = Byteable.Companion.byteToDateTime(buffer);
    }

    public String getUserName() {
        return userName;
    }

    public DateTime getBirthday() {
        return birthday;
    }

    public String getName() {
        return NAME;
    }


    @Override
    protected int getContentLength() {
        return Byteable.Companion.getStringLength(userName) + Byteable.Companion.getDateLength();
    }

    @Override
    protected void writeContent(ByteBuffer buffer) {
        buffer.put(Byteable.Companion.stringToByte(userName));
        buffer.put(Byteable.Companion.dateTimeToBytes(birthday));
    }

    @Override
    public String toString() {
        return "CreateUserMessage{" +
                "userName='" + userName + '\'' +
                ", birthday=" + birthday +
                '}';
    }
}

File: commonfamily/src/main/java/com/koenig/communication/messages/family/CreateUserMessage.kt
package com.koenig.communication.messages.family

import com.koenig.commonModel.Byteable
import com.koenig.commonModel.Component
import com.koenig.communication.messages.FamilyMessage

import org.joda.time.DateTime

import java.nio.ByteBuffer

/**
 * Created by Thomas on 11.10.2017.
 */

class CreateUserMessage : FamilyMessage {

    var userName: String
        private set
    var birthday: DateTime
        private set

    override val name: String
        get() = NAME


    override val contentLength: Int
        get() = Byteable.getStringLength(userName) + Byteable.dateLength

    constructor(userName: String, birthday: DateTime) : super(Component.FAMILY) {
        this.birthday = birthday
        this.userName = userName
    }

    constructor(version: Int, component: Component, fromId: String, toId: String, messageId: String, buffer: ByteBuffer) : super(version, component, fromId, toId, messageId) {
        userName = Byteable.byteToString(buffer)
        birthday = Byteable.byteToDateTime(buffer)
    }

    override fun writeContent(buffer: ByteBuffer) {
        buffer.put(Byteable.stringToByte(userName))
        buffer.put(Byteable.dateTimeToBytes(birthday))
    }

    override fun toString(): String {
        return "CreateUserMessage{" +
                "userName='" + userName + '\''.toString() +
                ", birthday=" + birthday +
                '}'.toString()
    }

    companion object {

        val NAME = "CreateUserMessage"
    }
}
